src/aws/nodemanager/nodemanager.py

```diff
@@ -170,6 +215,16 @@ def process_heartbeat(self, heartbeat: HeartBeatPacket):
             # TODO do something with the workers_stopped. These workers were running before. @Karan.
             self._tp.workers_running = heartbeat['workers_running']
             self._tp.workers_pending = heartbeat['workers_pending']
+            stopped_workers, new_workers = self._tp.worker_change()
+
+            # Add all tasks remaining in stopped worker assignments back to the taskpool
+            for worker in stopped_workers:
+                self._tp._tasks = self._tp._tasks + self._tp._task_assignment[worker]
```

Comment:
Doesn't `_tasks` contain all tasks? So adding the tasks of a stopped worker would duplicate these tasks, or am I wrong here?

Reply:
I have rearranged this. So _tasks contains all tasks which have not yet been assigned to a worker. THis way there are no duplicates

Reply:
so `_tasks` are all unassigned tasks, `_tasks_running` all running tasks (so also assigned), `_task_assignment` all tasks assigned but not yet running, and `_task_processing` tasks currently being processed?
What is the difference between `_tasks_running` and `_tasks_processing`?